build-cmd.sh

```diff
@@ -292,30 +292,40 @@ case "$AUTOBUILD_PLATFORM" in
             fi
         done
 
+	logical_cpus=$(sysctl hw.logicalcpu | cut -d ':' -d ' ' -f 2)
+
         sep "bootstrap"
         stage_lib="${stage}"/lib
         ./bootstrap.sh --prefix=$(pwd) --with-icu="${stage}"/packages
 
+	cat > "$top/$BOOST_SOURCE_DIR/tools/build/src/user-config.jam" <<EOF
+using clang-darwin : : :
+<architecture>combined <visibility>global <compileflags>"-arch x86_64 -arch arm64"
```

Comment:
I am currently failing to find the correct incantations to get boost to build all the correct assembly implementations of boost::context asm sources

https://paste.lindenlab.com/show/48345/

the current build produces an error like the above twice, both for x86_64 and arm64 as I understand the output

Reply:
note: the `<visibility>global` flag is intended to fix a bunch of warnings in the viewer build:
`Direct access in function 'boost::...' from file '.../build-darwin-x86_64/packages/lib/release/libboost_regex-mt-x64.a(cpp_regex_traits.o)' to global weak symbol 'guard variable for boost::cpp_regex_traits<char>::get_catalog_name_inst()::s_name' from file '.../build-darwin-x86_64/build/PROJECT_llfilesystem_TEST_lldiriterator.build/RelWithDebInfo/Objects-normal/x86_64/lldiriterator.o' means the weak symbol cannot be overridden at runtime. This was likely caused by different translation units being compiled with different visibility settings.`

Reply:
> I am currently failing to find the correct incantations to get boost to build all the correct assembly implementations of boost::context asm sources
> 
> the current build produces an error like the above twice, both for x86_64 and arm64 as I understand the output

Is aarch64 assembly the same as arm64? My surface reaction is that any new target platform requires new assembly code.